apm-agent-attach-cli/src/main/java/co/elastic/apm/attach/PgpSignatureVerifierLoader.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.attach;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystemNotFoundException;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class PgpSignatureVerifierLoader extends URLClassLoader {
+
+    private final String verifierImplementationName;
+
+    static PgpSignatureVerifierLoader getInstance(final String sourceLib, final Path targetLib, String verifierClassName) throws Exception {
+
+        // done lazily so that we create the logger only after proper initialization
+        final Logger logger = LogManager.getLogger(PgpSignatureVerifierLoader.class);
+
+        Path libPath;
+        // first try to see if the lib is available as a resource available to the current class loader
+        URL resourceUrl = PgpSignatureVerifierLoader.class.getResource(sourceLib);
```

Comment:
[question] why do we have to apply any fallback here ? Is that loading library is different when testing (BC is on the classpath), whereas when packaged we have a different behavior ?

Reply:
Add a comment that we provide both options to enable unit tests